packages/zql/src/ivm/constraint.test.ts

```diff
@@ -0,0 +1,352 @@
+              type: 'simple',
+              left: {type: 'column', name: 'id'},
+              op: '=',
+              right: {type: 'literal', value: 1},
+            },
+            {
+              type: 'simple',
+              left: {type: 'column', name: 'name'},
+              op: '=',
+              right: {type: 'literal', value: 'test'},
+            },
+          ],
+        } as const,
+        primary: ['id'] as const,
+        expected: {id: 1},
+      },
+      {
+        name: 'handles nested AND conditions',
+        condition: {
+          type: 'and',
+          conditions: [
+            {
+              type: 'simple',
+              left: {type: 'column', name: 'id'},
+              op: '=',
+              right: {type: 'literal', value: 1},
+            },
+            {
+              type: 'and',
+              conditions: [
+                {
+                  type: 'simple',
+                  left: {type: 'column', name: 'tenant'},
+                  op: '=',
+                  right: {type: 'literal', value: 'test'},
+                },
+              ],
+            },
+          ],
+        } as const,
+        primary: ['id', 'tenant'] as const,
+        expected: {id: 1, tenant: 'test'},
+      },
+      {
+        name: 'handles nested AND with OR conditions',
+        condition: {
+          type: 'and',
+          conditions: [
+            {
+              type: 'simple',
+              left: {type: 'column', name: 'id'},
+              op: '=',
+              right: {type: 'literal', value: 1},
+            },
+            {
+              type: 'and',
+              conditions: [
+                {
+                  type: 'simple',
+                  left: {type: 'column', name: 'tenant'},
+                  op: '=',
+                  right: {type: 'literal', value: 'test'},
+                },
+                {
+                  type: 'or',
+                  conditions: [
+                    {
+                      type: 'simple',
+                      left: {type: 'column', name: 'status'},
+                      op: '=',
+                      right: {type: 'literal', value: 'active'},
+                    },
+                    {
+                      type: 'simple',
+                      left: {type: 'column', name: 'status'},
+                      op: '=',
+                      right: {type: 'literal', value: 'pending'},
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        } as const,
+        primary: ['id', 'tenant'] as const,
+        expected: {id: 1, tenant: 'test'},
+      },
+    ])('$name', ({condition, primary, expected}) => {
+      const result = primaryKeyConstraintFromFilters(condition, primary);
+      expect(result).toEqual(expected);
+    });
+  });
+});
+
+test('returns undefined for operators other than equality', () => {
+  const operators: SimpleOperator[] = [
+    '>',
+    '<',
+    '>=',
```

Comment:
Can we not allow `>=`?

Reply:
Not sure. The existing constraint logic does look to create a `startAt` and scan from there so potentially we can.